### PR TITLE
ServerSentEventParser: Reducing allocations, indirections and removin…

### PR DESCRIPTION
**Pull request by viktorklang**
…g nested match

### CODE REVIEW

**Comment by viktorklang** [comment]
@hseeberger Slowly working my way towards net-negative ;-)

**Comment by viktorklang** [comment]
@hseeberger ping

File: akka-sse/src/main/scala/de/heikoseeberger/akkasse/client/ServerSentEventParser.scala [hunk 1]
```diff
@@ -60,10 +60,10 @@ private object ServerSentEventParser {
 
     def build(): ServerSentEvent =
       ServerSentEvent(
-        if (data.isEmpty) None else Some(data.mkString(LF)),
+        if (data.nonEmpty) Some(data.mkString("\n")) else None,
```
Comment (viktorklang) [inline akka-sse/src/main/scala/de/heikoseeberger/akkasse/client/ServerSentEventParser.scala]: I prefer testing for the generally successful case (branch prediction)

File: akka-sse/src/main/scala/de/heikoseeberger/akkasse/client/ServerSentEventParser.scala [hunk 2]
```diff
@@ -19,7 +19,7 @@ package client
 
 import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
 import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
-import scala.util.control.NonFatal
+import scala.util.Try
```
Comment (hseeberger) [inline akka-sse/src/main/scala/de/heikoseeberger/akkasse/client/ServerSentEventParser.scala]: Please remove unnecessary import.

Comment (viktorklang) [inline akka-sse/src/main/scala/de/heikoseeberger/akkasse/client/ServerSentEventParser.scala]: :+1:

**Comment by hseeberger** [comment]
Thanks!

**Pull request merged**